src/bin/cargo/commands/install.rs

```diff
@@ -133,6 +134,21 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
         .collect::<crate::CargoResult<Vec<_>>>()?;
 
     for (crate_name, _) in krates.iter() {
+        let package_name = PackageName::new(crate_name);
+        if !crate_name.contains("@") && package_name.is_err() {
+            for (idx, ch) in crate_name.char_indices() {
+                if !(unicode_xid::UnicodeXID::is_xid_continue(ch) || ch == '-') {
+                    let mut suggested_crate_name = crate_name.to_string();
+                    suggested_crate_name.insert_str(idx, "@");
+                    if let Ok((_, Some(_))) = parse_crate(&suggested_crate_name.as_str()) {
+                        return Err(anyhow!(
+                            "possible missing `@` detected in package name `{crate_name}`. Try `{suggested_crate_name}` instead."
```

Comment:
This does not read as an error message to me (saying what is wrong) but instead as a suggestion of what could be wrong and how that could be fixed which we usually put in `help:` sections of an error message.

I'd keep the original `package_name.unwrap_err()` and append something like this (note: I've not thought too deeply on this wording):
```

help: if this is meant to be a package name followed by a version, insert a `@` like `{suggested_crate_name}`
```

Reply:
I am traveling for a few days. Will try to finish this PR afterwards.

Reply:
Okay, had some time to follow up on this. The wording sounds pretty okay for me.